src/config_file.rs

```diff
@@ -1,175 +1,353 @@
+#[derive(Eq, PartialEq, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+#[serde(deny_unknown_fields)]
 pub struct Config {
-    pub file: Option<String>,
-    pub yaml: Option<Yaml>,
+    pub classic: Option<bool>,
+    pub blocks: Option<Vec<String>>,
+    pub color: Option<Color>,
+    pub date: Option<String>,
+    pub dereference: Option<bool>,
+    pub display: Option<Display>,
+    pub icons: Option<Icons>,
+    pub ignore_globs: Option<Vec<String>>,
+    pub indicators: Option<bool>,
+    pub layout: Option<Layout>,
+    pub recursion: Option<Recursion>,
+    pub size: Option<SizeFlag>,
+    pub sorting: Option<Sorting>,
+    pub no_symlink: Option<bool>,
+    pub total_size: Option<bool>,
+    pub symlink_arrow: Option<String>,
+}
+
+#[derive(Eq, PartialEq, Debug, Deserialize)]
+pub struct Color {
+    pub when: ColorOption,
+}
+
+#[derive(Eq, PartialEq, Debug, Deserialize)]
+pub struct Icons {
+    pub when: Option<IconOption>,
+    pub theme: Option<String>,
+}
+
+#[derive(Eq, PartialEq, Debug, Deserialize)]
+pub struct Recursion {
+    pub enabled: Option<bool>,
+    pub depth: Option<usize>,
+}
+
+#[derive(Eq, PartialEq, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Sorting {
+    pub column: Option<SortColumn>,
+    pub reverse: Option<bool>,
+    pub dir_grouping: Option<DirGrouping>,
 }
 
 impl Config {
-    /// This constructs a Config struct without a file [String] and without a [Yaml].
+    /// This constructs a Config struct with all None
     pub fn with_none() -> Self {
         Self {
-            file: None,
-            yaml: None,
-        }
-    }
-
-    /// This constructs a Config struct with a passed file [String] and without a [Yaml].
-    pub fn with_file(file: String) -> Self {
-        Self {
-            file: Some(file),
-            yaml: None,
+            classic: None,
+            blocks: None,
+            color: None,
+            date: None,
+            dereference: None,
+            display: None,
+            icons: None,
+            ignore_globs: None,
+            indicators: None,
+            layout: None,
+            recursion: None,
+            size: None,
+            sorting: None,
+            no_symlink: None,
+            total_size: None,
+            symlink_arrow: None,
         }
     }
 
-    /// This constructs a Config struct with a passed [Yaml] and without a file [String].
-    #[cfg(test)]
-    pub fn with_yaml(yaml: Yaml) -> Self {
-        Self {
-            file: None,
-            yaml: Some(yaml),
+    /// This constructs a Config struct with a passed file path [String].
+    pub fn with_file(file: String) -> Option<Self> {
+        match fs::read(&file) {
+            Ok(f) => Self::with_yaml(&String::from_utf8_lossy(&f)),
+            Err(e) => {
+                match e.kind() {
+                    std::io::ErrorKind::NotFound => {}
+                    _ => print_error!("Can not open config file {}: {}.", &file, e),
```

Comment:
actually, the filename is printed by `&file`.

the logic behind this error handing is that:
1. most of the `lsd` user has not configuration file currently
2. skip the not found error to avoid printing unnecessary error msg
3. if an error other than `FileNotFound` happened, print the filename and error msg

the problem may be that is users used a wrong file, and `lsd` has not found it, there will be no error msg

Reply:
I only wrote this comment here, because this is where the file information is lost in the flow of the configuration logic. My problem is that any other errors, caused by invalid configuration after this point, have no information any more in which file they happened.

Reply:
I have returned the error from `from_yaml` and print the error with the file name, when deserialized error.

Reply:
Yep, looks good to me.